src/forms/FieldsetTime.jsx

```diff
@@ -0,0 +1,247 @@
+		const currentTime = value ? LocalTime.parse(value) : LocalTime.of(0, 0, 0);
+		if (Number.isNaN(newValue)) {
+			// invalid value, so just force re-set to currentTime
+			onChange(currentTime.toString());
+			return;
+		}
+		// set the currentTime field corresponding to hour/minute
+		const adjustField = ADJUST_FIELD_MAP[fieldName];
+		const newTime = currentTime.with(adjustField, newValue);
+		if (is24Hr) {
+			onChange(newTime.toString());
+			return;
+		}
+		onChange(
+			newTime
+				.with(ChronoField.AMPM_OF_DAY, currentTime.get(ChronoField.AMPM_OF_DAY))
+				.toString()
+		);
+	}
+
+	// selects text when the hour or minute input gets focus
+	selectInputText(e: SyntheticEvent<HTMLInputElement>) {
+		e.currentTarget.select();
+	}
+
+	// handler for AM/PM change - trigger onChange callback with new hour offset
+	onMeridianChange = (e: SyntheticInputEvent<HTMLInputElement>) => {
+		if (!this.props.value || this.props.is24Hr) {
+			return;
+		}
+		const localTime = LocalTime.parse(this.props.value);
+		const meridian = parseInt(e.target.value, 10);
+		this.props.onChange(localTime.with(ChronoField.AMPM_OF_DAY, meridian).toString());
+	};
+
+	// Handle up/down arrow behavior in hour/minute fields
+	onNumberKeyDown = (e: SyntheticKeyboardEvent<HTMLInputElement>) => {
+		if (!KEY_INCREMENT[e.keyCode]) {
+			return;
+		}
+		e.preventDefault(); // don't do anything but what I tell you
+		const stepMultiplier = e.shiftKey ? 10 : 1; // hold shift for fast increment
+		const stepSize = KEY_INCREMENT[e.keyCode] * stepMultiplier;
+		const field = e.currentTarget.name; // 'hour' or 'minute'
+		const currentTime = this.props.value
+			? LocalTime.parse(this.props.value)
+			: LocalTime.of(0, 0, 0); // empty value is ok - just start incrementing from 00:00
+
+		const newTimeFields = {
+			hour: currentTime.hour(),
+			minute: currentTime.minute(),
+		};
+		newTimeFields[field] = newTimeFields[field] + stepSize;
+		// wrap the hour when minute reaches boundaries
+		if (newTimeFields.minute >= 60) {
+			newTimeFields.hour = newTimeFields.hour + 1;
+		}
+		if (newTimeFields.minute < 0) {
+			newTimeFields.hour = newTimeFields.hour - 1;
+		}
+
+		// trigger change with positive, modulated hour/minute values
+		const newTime = LocalTime.of(
+			(newTimeFields.hour + 24) % 24,
+			(newTimeFields.minute + 60) % 60,
+			0
+		);
+		this.props.onChange(newTime.toString());
+	};
+
+	render() {
+		const { id, error, disabled, name, is24Hr, value } = this.props;
+
+		const classNames = {
+			fauxInput: cx('fauxInput fauxInput--time', {
+				disabled,
+				error,
+			}),
+			meridian: cx(
+				MERIDIAN_INPUT_CLASS,
+				'flush--all border--none field--reset padding--left',
+				{
+					disabled,
+					error,
+				}
+			),
+		};
+		const displayId = id || name;
+
+		const { hour, minute, meridian } = getValueComponents(is24Hr, value);
+		return (
+			<fieldset className={classNames.fauxInput}>
+				<Flex noGutters>
+					<FlexItem shrink>
+						<input
```

Comment:
We should probably provide visually hidden `<label>`s for each input. This can be accomplished by adding the class `visibility--a11yHide`

Reply:
I'll clarify through docs, but this component is really intended to be a replacement specifically for `<input type="time" />` - labelling responsibilities are handled by parent, e.g. `InputTime`, which is generally the component that will be used by consumers.

Definitely need to make that clear in the code, though - for consistently, this component should _never_ be used directly in consumer code.

Reply:
In which case, I should probably just port the stories over to `InputTime` entirely, and not have `FieldsetTime.story.jsx` at all